File: Python/Program/Speaker.py
from KomponenEksternal import KomponenEksternal

class Speaker(KomponenEksternal):
    def __init__(self, merk="", nama="", harga=0, jenis_koneksi="", terhubung=False, daya_output=0, stereo=True):
        super().__init__(merk, nama, harga, jenis_koneksi, terhubung)
        self.daya_output = daya_output
        self.stereo = stereo
    
    # Setter
    def set_daya_output(self, daya_output):
        self.daya_output = daya_output
    
    def set_stereo(self, stereo):
        self.stereo = stereo
    
    # Getter
    def get_daya_output(self):
        return self.daya_output
    
    def get_stereo(self):
        return self.stereo
    
    # Tampilkan Info
    def tampilkan_info(self):
        return f"{super().tampilkan_info()} | Daya Output: {self.daya_output} Watt | Stereo: {'Ya' if self.stereo else 'Tidak'}"

File: Java/Program/Speaker.java
public class Speaker extends KomponenEksternal {
    private int dayaOutput;
    private boolean stereo;

    // Constructor
    public Speaker() {
        super();
        this.dayaOutput = 0;
        this.stereo = false;
    }

    public Speaker(String merk, String nama, int harga, String jenisKoneksi, boolean terhubung,
                  int dayaOutput, boolean stereo) {
        super(merk, nama, harga, jenisKoneksi, terhubung);
        this.dayaOutput = dayaOutput;
        this.stereo = stereo;
    }

    // Setter methods
    public void setDayaOutput(int dayaOutput) { this.dayaOutput = dayaOutput; }
    public void setStereo(boolean stereo) { this.stereo = stereo; }

    // Getter methods
    public int getDayaOutput() { return dayaOutput; }
    public boolean getStereo() { return stereo; }

    // Tampilkan Info
    @Override
    public String tampilkanInfo() {
        return super.tampilkanInfo() + " | Daya Output: " + dayaOutput + " Watt | Stereo: " + (stereo ? "Ya" : "Tidak");
    }
}